pkg/controller/bundle_sync_task.go

```diff
@@ -249,6 +298,46 @@ func (st *bundleSyncTask) isBundleReady() bool {
 	return true
 }
 
+// updateResourceCondition updates passed condition by fetching information from an existing resource condition if present.
+// Sets LastTransitionTime to now if the status has changed.
+// Returns true if resource condition in the bundle does not match and needs to be updated.
+func updateResourceCondition(b *smith_v1.Bundle, resName smith_v1.ResourceName, condition *smith_v1.ResourceCondition) bool {
```

Comment:
This should like it updatesResourceConditions, but it just checks. shouldUpdateResourceCondition? hasResourceConditionChanged?